lib/pure/strutils.nim

```diff
@@ -154,6 +154,124 @@ proc isDigit*(c: char): bool {.noSideEffect,
     doAssert isDigit('8') == true
   return c in Digits
 
+proc isNumeric*(s: string, enableNaNInf, enableLooseDot = false): bool =
+  ## Checks whether the string is numeric.
+  ## When the string is an integer, float or exponential, it returns true,
+  ## otherwise it returns false.
+  ## The `enableNaNInf` value indicates whether the `NaN` and `Inf` values
+  ## are valid.
+  ## The `enableLooseDot` value indicates whether loose point values such as
+  ## `.9` and `9.` are valid. 
+  ##
+  ## **Note:** The reason that `parseFloat()` is not used to achieve this is its
+  ## poor performance.
+  runnableExamples:
+    doAssert isNumeric("123") == true
```

Comment:
doAssert isNumeric("123") == true
=>
doAssert isNumeric("123")

doAssert not isNumeric("e123.45") == true
=>
doAssert not isNumeric("e123.45")

(+everywhere relevant)